src/services/hooks.rs

```diff
@@ -1,23 +1,27 @@
-use crate::utils::context::Context;
+use crate::utils::context::ContextCache;
 use crate::models::backend::webhooks::{Payload, PayloadDetails};
 use anyhow::Result;
+use crate::utils::cache::Cache;
 
-pub fn invalidate_caches(context: &Context, payload: &Payload) -> Result<()> {
+pub fn invalidate_caches(context: &impl ContextCache, payload: &Payload) -> Result<()> {
```

Comment:
I am not the biggest fan of the `ContextCache`. Either I would say we make the whole `Context` into a trait (might be too much right now) or we just pass in `Cache` here.

If neither is easy to do right now we can stick with the `ContextCache`

Reply:
Maybe the latter. The definition of `Context` :

```rust
pub struct Context<'a, 'r> {
    request: &'a Request<'r>,
    cache: ServiceCache,
}
```

It is coupled with the request. Ideally we have it as a trait, which could be a bit too much of a change now. Not sure what you have against though having `ContextCache`. I feel is more behaviour-driven. There is a `cache` aspect and a `request` aspect of the context, that we can selectively use.